File: ios/Plugin/TelephonyPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(NetworkQualityPlugin, "NetworkQuality",
           CAP_PLUGIN_METHOD(getInfo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getRadioInfo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getNetworkType, CAPPluginReturnPromise);
)

File: ios/Plugin/TelephonyPlugin.swift
import Foundation
import Capacitor

@objc(NetworkQualityPlugin)
public class NetworkQualityPlugin: CAPPlugin {
    @objc func getInfo(_ call: CAPPluginCall) {
        call.reject("Not available on iOS")
    }

    @objc func getRadioInfo(_ call: CAPPluginCall) {
        call.reject("Not available on iOS")
    }

    @objc func getNetworkType(_ call: CAPPluginCall) {
        call.reject("Not available on iOS")
    }
}
